tests/lifecycleHooks.lua

```diff
@@ -0,0 +1,241 @@
+local TestEZ = require(script.Parent.Parent.TestEZ)
+
+local function expectShallowEquals(array1, array2)
+	local function shallowEquals()
+		-- shallow equals between lifecycleOrder and passed array
+		for index, value in ipairs(array1) do
+			if array2[index] ~= value then
+				return false
+			end
+		end
+
+		for index, value in ipairs(array2) do
+			if array1[index] ~= value then
+				return false
+			end
+		end
+
+		return true
+	end
+
+	assert(
+		shallowEquals(),
+		string.format("lifecycle order did not match expected order.\nGot: {\n\t%s\n}", table.concat(array1, "\n\t"))
+	)
```

Comment:
The second value of `assert` isn't lazily evaluated. If we're going to do string formatting work, we should branch on our result instead of using `assert`.